prow/cmd/sinker/main.go

```diff
@@ -105,17 +105,24 @@ func (c *controller) clean() {
 		c.logger.WithError(err).Error("Error listing prow jobs.")
 		return
 	}
+
+	// Only delete pod if its prowjob is marked as finished
+	isFinished := make(map[string]bool)
```

Comment:
I assume we are correctly marking prowJobs as finished for any terminal state? We *need* to not pile up dead pods. My concern is that the previous code looked reasonable for deleting terminal pods ~30 minutes after they finish at which point I would expect plank to have already marked them completed, but this suggests that this was not true...

Reply:
plank works by polling k8s every 30s. It's possible that during that interval sinker may delete a complete pod for a prowjob that is yet to be marked as complete by plank.

Reply:
Ok SGTM, thanks! WDYT about the nits above being fixed in a follow-up? (sen is in GMT+8? and we should roll this out ~ASAP if we can).

Reply:
Sure, np